### PR TITLE
Fix erroring of slugid implementation, migrate tests from tc-client

### LINKED ISSUE #19: `slugid()` throws TypeError
https://github.com/taskcluster/taskcluster-client-web/blob/836588944e4f0f2cc90ac24ab3543b7efd369562/src/utils.js#L87

Nightly gives me "TypeError: 0 is read-only", Chrome does "TypeError: Cannot assign to read only property '0' of string '8a170ed0-4a2d-468f-8847-d446988abcb5'

### PR DESCRIPTION
**Pull request by eliperelman**
Fixes #19.

### CODE REVIEW

**Comment by eliperelman** [comment]
In a nutshell, there were 2 problems when this was migrated from tc-client:

1. The UUID implementation returned a string instead of a buffer, meaning no valid bitwise operations were actually going to happen.
2. We were base64-ing the buffer itself instead of the stringified version of it.

File: test/slugid_test.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import { nice, v4 } from '../src';
+
+/**
+ * `spreader` runs a test against the `generator` function, to check that
+ * when calling it 64*40 times, the range of characters per string position it
+ * returns matches the array `actual`, where each entry in `expected` is a
+ * string of all possible characters that should appear in that position in the
+ * string, at least once in the sample of 64*40 responses from the `generator`
+ * function */
+const spreader = (generator) => {
```
Comment (djmitche) [inline test/slugid_test.js]: wow, this is cool

Comment (eliperelman) [reply test/slugid_test.js]: I can't take credit for this, I think this was all @jonasfj (ripped and modified from `slugid`).

**Comment by eliperelman** [comment]
Published in v5.0.1.

**Pull request merged**